src/beliefs.jl

```diff
@@ -93,14 +93,18 @@ ParticleCollection(p::AbstractVector{T}) where T = ParticleCollection{T}(p, noth
 
 n_particles(b::ParticleCollection) = length(b.particles)
 particles(p::ParticleCollection) = p.particles
-weights(b::ParticleCollection) = ones(n_particles(b))
+weights(b::ParticleCollection) = fill(1/n_particles(b),n_particles(b))
```

Comment:
Good catch! However, `weights` does not need to return normalized weights. I would vote for keeping this `ones(n_particles(b))` and implement `weight_sum(b::ParticleCollection) = n_particles(b))` instead.

Reply:
I agree with @lassepe 's take on this

Reply:
That's also fine, as long as all weight functions are consistent, i.e. also implementing `weight(b::ParticleCollection, i::Int) = 1.0` and `weighted_particles(p::ParticleCollection) = (s=>1.0 for s in p.particles)`. I will take care of that.